resources/views/layouts/partials/light_dark_mode_logic.blade.php

```diff
@@ -1,40 +1,74 @@
 <script>
-    const defaultColorMode = window.localStorage.getItem('mode')
-        ? window.localStorage.getItem('mode')
-        : '{{ backpack_theme_config('options.defaultColorMode') ?? 'light' }}';
-    const body = document.getElementsByTagName('body')[0];
-    const html = document.getElementsByTagName('html')[0];
-
-    // Allow users to subscribe to an event that listens for changes in the color mode
-    // and adjust what's been displayed as needed (like images, etc)
-    colorMode = {
-        value: null,
-        set(theme) {
-            const previousTheme = theme === 'light'
-                ? 'dark'
-                : 'light';
-
-            html.dataset.bsTheme = theme;
-            body.classList.remove('theme-' + previousTheme);
-            body.classList.add('theme-' + theme);
-
-            window.localStorage.setItem('mode', theme);
-
-            this.value = theme;
-            this.colorModeListener(theme);
-        },
+    class ColorMode {
+        constructor(states, defaultColorMode) {
+            this.value = null;
+            this.valueSystem = null;
+            this.listeners = [];
+            this.states = states;
+
+            this.set(window.localStorage.getItem('colorMode') ?? defaultColorMode);
+
+            // listen for color scheme changes
+            const query = window.matchMedia('(prefers-color-scheme: dark)');
```

Comment:
This fixes https://github.com/Laravel-Backpack/theme-tabler/issues/62

GJ @promatik